### PR TITLE
core: remove untraceable dummy conflicting transactions

### PR DESCRIPTION
**Pull request by AnnaShaleva**

### CODE REVIEW

**Review by roman-khimov — APPROVED** [review]

File: pkg/core/dao/dao.go [hunk 1]
```diff
@@ -616,9 +615,31 @@ func (dao *Simple) StoreAsCurrentBlock(block *block.Block, buf *io.BufBinWriter)
...
+func (dao *Simple) StoreAsTransaction(tx *transaction.Transaction, index uint32, p2pSigExtensions bool, buf *io.BufBinWriter) error {
+	err := dao.storeTransaction(tx, index, buf)
+	if err != nil {
+		return err
+	}
+	if p2pSigExtensions {
+		for _, attr := range tx.GetAttributes(transaction.ConflictsT) {
+			buf.Reset()
+			hash := attr.Value.(*transaction.Conflicts).Hash
+			dummyTx := transaction.NewTrimmedTX(hash)
```
Comment (fyrchik) [inline pkg/core/dao/dao.go:630]: So we allocate a struct only to encode 1 constant byte after? Key may also be reused.

**Comment by AnnaShaleva** [comment]
```
name                  old time/op    new time/op    delta
StoreAsTransaction-8    1.71µs ± 9%    1.16µs ± 2%  -31.74%  (p=0.000 n=10+8)

name                  old alloc/op   new alloc/op   delta
StoreAsTransaction-8    1.73kB ± 0%    1.02kB ± 0%  -40.74%  (p=0.000 n=10+10)

name                  old allocs/op  new allocs/op  delta
StoreAsTransaction-8      29.0 ± 0%      21.0 ± 0%  -27.59%  (p=0.000 n=10+10)
```

File: pkg/core/dao/dao.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/core/dao/dao.go]: Fixed, benchmark results for these approaches are in https://github.com/nspcc-dev/neo-go/pull/2134#issuecomment-900975777.

File: pkg/core/dao/dao.go [hunk 2]
```diff
@@ -556,7 +555,7 @@ func (dao *Simple) StoreAsBlock(block *block.Block, buf *io.BufBinWriter) error
 }
 
 // DeleteBlock removes block from dao.
-func (dao *Simple) DeleteBlock(h util.Uint256, w *io.BufBinWriter) error {
+func (dao *Simple) DeleteBlock(h util.Uint256, p2pSigExtensions bool, w *io.BufBinWriter) error {
```
Comment (roman-khimov) [inline pkg/core/dao/dao.go:558]: We can hide `p2pSigExtensions` in `Simple` itself, `NewSimple` can take it as an argument.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**